src/main/java/com/algorand/algosdk/v2/client/common/AlgodClient.java

```diff
@@ -133,4 +137,39 @@ public GetPendingTransactions GetPendingTransactions() {
     public PendingTransactionInformation PendingTransactionInformation(String txid) {
         return new PendingTransactionInformation((Client) this, txid);
     }
+
+    /**
+     * Given a application id, it returns application information including creator,
+     * approval and clear programs, global and local schemas, and global state.
+     * /v2/applications/{application-id}
+     */
+    public GetApplicationByID GetApplicationByID(Long applicationId) {
+        return new GetApplicationByID((Client) this, applicationId);
+    }
+
+    /**
+     * Given a asset id, it returns asset information including creator, name, total
+     * supply and special addresses.
+     * /v2/assets/{asset-id}
+     */
+    public GetAssetByID GetAssetByID(Long assetId) {
+        return new GetAssetByID((Client) this, assetId);
+    }
+
+    /**
+     * Given TEAL source code in plain text, return base64 encoded program bytes and
+     * base32 SHA512_256 hash of program bytes (Address style).
+     * /v2/teal/compile
+     */
+    public TealCompile TealCompile() {
+        return new TealCompile((Client) this);
+    }
+
+    /**
+     * /v2/teal/dryrun
+     */
+    public TealDryRun TealDryRun() {
```

Comment:
What do you think about having a constructor that accepts the body parameter? I guess we would need multiple for this (for the RawTransaction endpoint):

```
TealDryRun(){ return new TealDryRun((client) this); }
TealDryRun(DryRunParams params){ return new TealDryRun((client)this).params(params); }
```

Reply:
This probably isn't worth worrying about right now.